orttraining/orttraining/python/training/ortmodule/torch_cpp_extensions/cpu/aten_op_executor/aten_op_executor.cc

```diff
@@ -70,7 +95,20 @@ struct ATenOperator {
     c10::IValue i_value;
     // Create the torch tensor from this DLPack no matter we need it or not below,
     // so that the dlpack's deleter will be triggered when torch tensor is out of scope.
-    at::Tensor tensor = at::fromDLPack(dlpack);
+    at::Tensor tensor;
+    if (dlpack->dl_tensor.data){
+      tensor = at::fromDLPack(dlpack);
+    } else {
+      // sometimes the tensor does not contain any data. eg: the maxindices output 
+      // for embedding_bag_backward cuda kernel. In such cases, fromDLPack() errors out
+      // while trying to make a tensor while trying to match the data's device with 
+      // the device specified in dlpack.
+      // so skipping using fromDLPack in such cases to create empty tensor.
+      at::Device device = getATenDevice(dlpack->dl_tensor.device);
+      at::ScalarType stype = at::toScalarType(dlpack->dl_tensor.dtype);
+      tensor = at::empty(at::IntArrayRef(dlpack->dl_tensor.shape, dlpack->dl_tensor.ndim), 
+                         at::device(device).dtype(stype));
+    }
```

Comment:
You may need to call "dlpack->deleter(const_cast<DLManagedTensor*>(dlpack));" here, otherwise the dlpack will leak.